cpp/src/replace/replace.cu

```diff
@@ -87,6 +87,169 @@ __device__ auto get_new_value(cudf::size_type idx,
   return thrust::make_pair(new_value, output_is_valid);
 }
 
+__device__ int get_new_string_value(cudf::size_type idx,
+                                     cudf::column_device_view& input,
+                                     cudf::column_device_view& values_to_replace,
+                                     cudf::column_device_view& replacement_values) {
+  cudf::string_view inputString = input.element<cudf::string_view>(idx);
+  int match = -1;
+  for (int i = 0; i < values_to_replace.size(); i++) {
```

Comment:
First, could this be:

```
for (auto const& value_string : values_to_replace) {
    if (input_string == value_string)
        match = i;
}
```

Second, shouldn't we `break` when a match is found?

Reply:
It does need the value of the index i to work, so I don't think iterating over it as you describe would work. I did add in a break to terminate early.